include/deal.II/numerics/data_out.h

```diff
@@ -429,7 +429,7 @@ class DataOut : public DataOut_DoFData<DoFHandlerType,
   virtual cell_iterator
   next_cell(const cell_iterator &cell);
 
-private:
+protected:
```

Comment:
I don't think you need this: Even in derived classes, you shouldn't have to call these function objects yourself. I would much rather you leave these objects `private` and instead introduced a function `get_cell_selection()` that returns copies of the two function objects as a `std::pair`. Then you can call them from hand without having to know the internals of the class.